packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -136,16 +148,13 @@ export default util.createRule<Options, MessageId>({
       },
     };
 
-    function isHigherPrecedenceThanAwait(node: ts.Node): boolean {
+    function isHigherPrecedenceThanUnary(node: ts.Node): boolean {
       const operator = tsutils.isBinaryExpression(node)
         ? node.operatorToken.kind
         : ts.SyntaxKind.Unknown;
       const nodePrecedence = util.getOperatorPrecedence(node.kind, operator);
-      const awaitPrecedence = util.getOperatorPrecedence(
-        ts.SyntaxKind.AwaitExpression,
-        ts.SyntaxKind.Unknown,
-      );
-      return nodePrecedence > awaitPrecedence;
+      const unaryPrecedence = OperatorPrecedence.Unary;
+      return nodePrecedence > unaryPrecedence;
```

Comment:
Nit: I don't see a strong reason to make a variable here, given that its name is a subset of its value's words?

```suggestion
      return nodePrecedence > OperatorPrecedence.Unary;
```

Reply:
You are right. Fixed!